customer-terminate-service/app/src/test/java/se/inera/intyg/cts/infrastructure/service/CreateEncryptedZipPackageTest.java

```diff
@@ -0,0 +1,208 @@
+import org.junit.jupiter.api.Test;
+import se.inera.intyg.cts.domain.model.Password;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.infrastructure.persistence.entity.CertificateEntity;
+import se.inera.intyg.cts.infrastructure.persistence.entity.CertificateTextEntity;
+import se.inera.intyg.cts.infrastructure.persistence.entity.TerminationEntity;
+import se.inera.intyg.cts.infrastructure.persistence.repository.InMemoryCertificateEntityRepository;
+import se.inera.intyg.cts.infrastructure.persistence.repository.InMemoryCertificateTextsEntityRepository;
+import se.inera.intyg.cts.infrastructure.persistence.repository.InMemoryTerminationEntityRepository;
+
+class CreateEncryptedZipPackageTest {
+
+  public static final String DESTINATION_DIRECTORY = "./destination_directory";
+  private InMemoryTerminationEntityRepository inMemoryTerminationEntityRepository;
+  private InMemoryCertificateEntityRepository inMemoryCertificateEntityRepository;
+  private InMemoryCertificateTextsEntityRepository inMemoryCertificateTextEntityRepository;
+  private CreateEncryptedZipPackage createEncryptedZipPackage;
+
+  private Termination termination;
+  private TerminationEntity terminationEntity;
+  private List<CertificateEntity> certificates;
+  private List<CertificateTextEntity> certificateTexts;
+
+  private final Password password = new Password("thisismypassword");
+  private File zipFile;
+
+  @BeforeEach
+  void setUp() {
+    inMemoryTerminationEntityRepository = new InMemoryTerminationEntityRepository();
+    inMemoryCertificateEntityRepository = new InMemoryCertificateEntityRepository();
+    inMemoryCertificateTextEntityRepository = new InMemoryCertificateTextsEntityRepository();
+
+    createEncryptedZipPackage = new CreateEncryptedZipPackage(
+        inMemoryTerminationEntityRepository,
+        inMemoryCertificateEntityRepository,
+        inMemoryCertificateTextEntityRepository
+    );
+
+    termination = defaultTermination();
+    terminationEntity = defaultTerminationEntity();
+    inMemoryTerminationEntityRepository.save(terminationEntity);
+
+    certificates = certificateEntities(terminationEntity, 20, 5);
+    inMemoryCertificateEntityRepository.saveAll(certificates);
+
+    certificateTexts = certificateTextEntities(terminationEntity, 10);
+    inMemoryCertificateTextEntityRepository.saveAll(certificateTexts);
+  }
+
+  @AfterEach
+  void tearDown() {
+    if (zipFile != null) {
+      try {
+        Files.deleteIfExists(Path.of(zipFile.getPath()));
+        if (Files.exists(Path.of(DESTINATION_DIRECTORY))) {
+          Files.walk(Path.of(DESTINATION_DIRECTORY))
+              .sorted(Comparator.reverseOrder())
+              .map(Path::toFile)
+              .forEach(File::delete);
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  @Test
+  void shallReturnZipFileNamedWithHsaId() {
+    zipFile = createEncryptedZipPackage.create(termination, password);
```

Comment:
The only difference from mocks and InMemory-classes is that instead of mocking I prepare the inMemory-classes with required data and can validate that the result is as expected. I think it is a good thing that the test doesn't have to mock the behaviours that the test is dependent on.